javascript/javascript1/week2/formal-fullname.js

```diff
@@ -0,0 +1,29 @@
+
+
+function getFullname (firstname, surname, useFormalName, gender) {
+ if (firstname && surname)  {
```

Comment:
This is a great oppotunity to avoid unnecessary nesting of `if` conditionals by inverting the logic:
```js
function getFullname (firstname, surname, useFormalName, gender) {
  if (!firstname || !surname)  { // inverted logic
     return (`"Please fill up firstname and surname."`)
  }
  // avoids otherwise needed if inside if.

  ...
```
vs.

```js
function getFullname (firstname, surname, useFormalName, gender) {
  if (firstname && surname)  {
    // Now code must be located inside the if condition
     ...
  }
  else {
     return (`"Please fill up firstname and surname."`)
  }
```